File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/Prompt.java
package Auxiliary_Modules;

public class Prompt {
    public AXKeyValuePair kv = new AXKeyValuePair();
    String prompt = "";
    String regex = "";
    public Prompt() {
        kv.setKey("default");
    }

    public String getPrompt() {
        return prompt;
    }

    public void setPrompt(String prompt) {
        this.prompt = prompt;
    }
    public Boolean process(String in1){
        kv.setValue(RegexUtil.extractRegex(regex,in1));
        return kv.getValue().isEmpty(); // is prompt still active?
    }

    public AXKeyValuePair getKv() {
        return kv;
    }

    public void setRegex(String regex) {
        this.regex = regex;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/Prompt.kt
package auxiliary_modules

class Prompt {
    var kv = AXKeyValuePair()
    var prompt = ""
    private var regex = ""

    init {
        kv.key = "default"
    }

    fun process(in1: String): Boolean {
        kv.value = RegexUtil.extractRegex(regex, in1)
        return kv.value.isEmpty() // is prompt still active?
    }

    fun setRegex(regex: String) {
        this.regex = regex
    }
}